dask_expr/expr.py

```diff
@@ -1092,6 +1092,18 @@ class ExplodeSeries(Blockwise):
 class ExplodeFrame(ExplodeSeries):
     _parameters = ["frame", "column"]
 
+    def _simplify_up(self, parent):
+        if isinstance(parent, Projection):
+            columns = set(parent.columns).union(self.column)
+            if columns == set(self.frame.columns):
+                # Don't add unnecessary Projections, protects against loops
+                return
+
+            return type(parent)(
+                type(self)(self.frame[sorted(columns)], *self.operands[1:]),
+                *parent.operands[1:],
+            )
```

Comment:
Parent is a Projection here right?  Maybe `type(parent)` isn't necessary, but we can just use `[...]` (implicitly, I'm not expecting any relevant subclasses of Projection).

Also, does explode add columns?  If not, then maybe we don't need the outer projection at all?

Reply:
We have the type thing in a couple of places, I can clean them up all at once when the simplify prs are in.

No it does not, but you might have something like the following:

```
df.explode(column="a")["b"]
```

You have to keep "a" and "b" before applying explode, that's why we need both projections unfortunately.